lir/algorithms/bootstraps.py

```diff
@@ -0,0 +1,174 @@
+    samples.
+
+    Computing these intervals is done by creating interpolation functions that map the best estimate to the
+    difference between the best estimate and the lower and upper bounds of the confidence interval. To achieve this,
+    two subclasses are provided that differ in how the data points for interval estimation are obtained.
+
+    - BootstrapAtData: Uses the original training data points for interval estimation.
+    - BootstrapEquidistant: Uses equidistant points within the range of the training data for interval estimation.
+
+    The AtData variant allows for more complex data types, while the Equidistant variant is only suitable for continuous
+    features.
+
+    Attributes:
+        steps: The steps of s pipeline to be bootstrapped.
+        n_bootstraps: int: The number of bootstrap samples to generate.
+        interval: tuple[float, float]: The lower and upper quantiles for the confidence interval.
+        seed: int | None: The random seed for reproducibility.
+        n_points: int | None: Number of equidistant points to use for interval estimation (BootstrapEquidistant only).
+
+        f_delta_interval_lower: Interpolation function for the lower bound of the interval.
+        f_delta_interval_upper: Interpolation function for the upper bound of the interval.
+    """
+
+    def __init__(
+        self,
+        steps: list[tuple[str, Any]],
+        n_bootstraps: int = 400,
+        interval: tuple[float, float] = (0.05, 0.95),
+        seed: int | None = None,
+        n_points: int | None = 1000,
+    ):
+        """Initialize the TrainDataBootstrap with the given pipeline steps, number of bootstraps, and interval.
+
+        Parameters:
+        param steps: list[tuple[str, Any]]: The steps of the pipeline to be bootstrapped.
+        param n_bootstraps: int: The number of bootstrap samples to generate. Default is 400.
+        param interval: tuple[float, float]: The lower and upper quantiles for the confidence interval.
+                                             Default: (0.05,0.95).
+        param n_points: int | None: The number of equidistant points to use for interval estimation. Default is 1000.
+                                    If None, uses the number of instances in the training data.
+        """
+        super().__init__(steps)
+
+        self.interval = interval
+        self.n_bootstraps = n_bootstraps
+        self.seed = seed
+        self.n_points = n_points
+
+        self.f_delta_interval_lower = None
+        self.f_delta_interval_upper = None
+
+    @abstractmethod
+    def get_bootstrap_data(self, instances: FeatureData) -> FeatureData:
+        """Get the data points to use for interval estimation.
+
+        param instances: FeatureData: The feature data to fit the bootstrap system on.
+        return FeatureData: The feature data to use for interval estimation.
+        """
+        pass
```

Comment:
Per conventie (zie https://docs.python.org/3/library/exceptions.html#NotImplementedError)

```suggestion
        raise NotImplementedError
```